frontend/src/concepts/modelRegistry/context/ModelRegistrySelectorContext.tsx

```diff
@@ -20,52 +20,66 @@ export const ModelRegistrySelectorContext = React.createContext<ModelRegistrySel
   updatePreferredModelRegistry: () => undefined,
 });
 
-export const ModelRegistrySelectorContextProvider =
-  conditionalArea<ModelRegistrySelectorContextProviderProps>(
-    SupportedArea.MODEL_REGISTRY,
-    true,
-  )(({ children }) => {
-    const [modelRegistries, isLoaded, error] = useModelRegistries();
-    const [preferredModelRegistry, setPreferredModelRegistry] =
-      React.useState<ModelRegistrySelectorContextType['preferredModelRegistry']>(undefined);
-
-    const firstModelRegistry = modelRegistries.length > 0 ? modelRegistries[0] : null;
+export const ModelRegistrySelectorContextProvider: React.FC<
+  ModelRegistrySelectorContextProviderProps
+> = ({ children, ...props }) => {
+  if (useModelRegistryEnabled()) {
+    return (
+      <EnabledModelRegistrySelectorContextProvider {...props}>
+        {children}
+      </EnabledModelRegistrySelectorContextProvider>
+    );
+  }
+  return children;
+};
 
-    React.useEffect(() => {
-      if (firstModelRegistry && !preferredModelRegistry) {
-        setPreferredModelRegistry(firstModelRegistry);
-      }
-    }, [firstModelRegistry, preferredModelRegistry]);
+export const EnabledModelRegistrySelectorContextProvider: React.FC<
+  ModelRegistrySelectorContextProviderProps
+> = ({ children }) => {
+  const [modelRegistries, isLoaded, error] = useModelRegistries();
+  const [preferredModelRegistry, setPreferredModelRegistry] =
+    React.useState<ModelRegistrySelectorContextType['preferredModelRegistry']>(undefined);
 
-    const updatePreferredModelRegistry = React.useCallback<
-      ModelRegistrySelectorContextType['updatePreferredModelRegistry']
-    >((modelRegistry) => {
-      setPreferredModelRegistry(modelRegistry);
-    }, []);
+  const firstModelRegistry = modelRegistries.length > 0 ? modelRegistries[0] : null;
 
-    if (!isLoaded) {
-      return <Bullseye>Loading model registries...</Bullseye>;
+  React.useEffect(() => {
+    if (firstModelRegistry && !preferredModelRegistry) {
+      setPreferredModelRegistry(firstModelRegistry);
     }
+  }, [firstModelRegistry, preferredModelRegistry]);
 
-    if (error) {
-      return (
-        <Bullseye>
-          <Alert title="Model registry load error" variant="danger" isInline>
-            {error.message}
-          </Alert>
-        </Bullseye>
-      );
-    }
+  const updatePreferredModelRegistry = React.useCallback<
+    ModelRegistrySelectorContextType['updatePreferredModelRegistry']
+  >((modelRegistry) => {
+    setPreferredModelRegistry(modelRegistry);
+  }, []);
```

Comment:
Is this necessary, doesn't `setPreferredModelRegistry` satisfy the type for the context?

Reply:
Everything inside `EnabledModelRegistrySelectorContextProvider` is existing code, just the whitespace changed because I removed the wrapping `conditionalArea`. I could improve this while I'm in here but I wonder if that's outside the scope of this PR?

Reply:
I'll just keep improvements to the preexisting code here in a separate commit in case we need to revert them I think.

Reply:
I think any time we touch code it's fair to also improve it. 
But I'll leave it to you if you choose to do something in this PR or capture it for later.

Reply:
The `set` function returned from `React.useState` is considered stable.